test/order_test.rb

```diff
@@ -111,13 +111,37 @@
       expect(order.total).must_equal before_total
     end
   end
+
+  describe "#remove_product" do
+    it "removes a product from the products array" do
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      before_count = products.count
+
+      order = Order.new(1337, products, customer)
+      product = "banana"
+
+      order.remove_product(product)
+      expected_count = (before_count-1)
+
+      expect(order.products.count).must_equal expected_count
+    end
+  end
 end
 
 # TODO: change 'xdescribe' to 'describe' to run these tests
-xdescribe "Order Wave 2" do
+describe "Order Wave 2" do
   describe "Order.all" do
     it "Returns an array of all orders" do
-      # TODO: Your test code here!
+      orders = Order.all
+
+      expect(orders.length).must_equal 100
+      orders.each do |o|
```

Comment:
I'm not a huge fan of `o` as the iteration variable here. Single letter variables aren't descriptive in most cases, and are easier to misread than words.